### PR TITLE
PIM-6805 - Remove variant group (technical stuff)

### PR DESCRIPTION
**Pull request by aRn0D**
**Description (for Contributor and Core Developer)**

- [x] Rename SiblingUniqueVariantAxesValidator
- [x] Integration tests
- [x] Behat
- [x] Change log

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductRepository.php [hunk 1]
```diff
@@ -87,16 +87,6 @@ public function getItemsFromIdentifiers(array $identifiers)
     /**
      * {@inheritdoc}
      */
-    public function findAllForVariantGroup(GroupInterface $variantGroup, array $criteria = [])
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductRepository.php]: you can remove setProductQueryBuilderFactory, setGroupRepository and setReferenceDataRegistry methods too

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ValueController.php [hunk 2]
```diff
@@ -21,7 +21,7 @@
  * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
-class ProductTemplateController
+class ValueController
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/ValueController.php]: I don't understand that, can you explain please?

Comment (aRn0D) [reply src/Pim/Bundle/EnrichBundle/Controller/Rest/ValueController.php]: This controller is only used to validate some values when you mass edit common attribute nothing related to the product template.

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductRepository.php (hunk 1 above)
Comment (aRn0D) [reply src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductRepository.php]: I am waiting for the CI, if it is ok I will merge and open a small PR to fix it.

**Pull request merged**